### PR TITLE
CORE-6192 - remove-persistence-module

### PR DESCRIPTION
**Pull request by driessamyn**
- Removed `MappedSchema`
- Moved `CordaPersistenceException` and `UUIDConverter` into application module
- Removed persistence module
- Fixed up obsolete ledger module to support these changes.

Runtime os PR: https://github.com/corda/corda-runtime-os/pull/1890

### CODE REVIEW

File: ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/persistence/MappedSchema.kt [hunk 1]
```diff
@@ -1,6 +1,9 @@
-package net.corda.v5.persistence
+package net.corda.v5.ledger.obsolete.persistence
 
 /**
+ * NOTE: MOVED THIS HERE AS THIS CLASS IS NOW REDUNDANT IN THE API. (CORE-6192)
```
Comment (driessamyn) [inline ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/persistence/MappedSchema.kt:4]: FYI @vlajos 
Moved this class into this module as it is not needed anymore elsehwere.

Comment (lankydan) [reply ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/persistence/MappedSchema.kt:4]: How will developers create custom tables in C5?

Before we would use `MappedSchema` for both ledger and non-ledger tables. With this change, I think we would lose that.

Comment (lankydan) [reply ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/persistence/MappedSchema.kt:4]: Discussed with @driessamyn, due to CPK metadata now knowing what entities we have as well as liquibase scripts executing as defined in a CPK, we no longer need this class to create custom tables of any type.

**Review by lankydan — APPROVED** [review]

File: ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/persistence/MappedSchema.kt (hunk 1 above)
Comment (driessamyn) [reply ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/persistence/MappedSchema.kt:4]: Classes annotated with `@Entity` and `@CordaSerializable` are listed in the `Corda-Entity-Classes` CPK manifest attribute. This is what is used when creating a DB Sandbox.

**Review by ronanbrowne — APPROVED** [review]
LGTM

**Pull request merged**